src/core_codemods/combine_isinstance_issubclass.py

```diff
@@ -0,0 +1,99 @@
+import libcst as cst
+from libcst import matchers as m
+
+from codemodder.codemods.utils import extract_boolean_operands
+from codemodder.codemods.utils_mixin import NameResolutionMixin
+from core_codemods.api import Metadata, ReviewGuidance, SimpleCodemod
+
+
+class CombineIsinstanceIssubclass(SimpleCodemod, NameResolutionMixin):
+    metadata = Metadata(
+        name="combine-isinstance-issubclass",
+        summary="Simplify Boolean Expressions Using `isinstance` and `issubclass`",
+        review_guidance=ReviewGuidance.MERGE_WITHOUT_REVIEW,
+        references=[],
+    )
+    change_description = "Use tuple of matches instead of boolean expression with `isinstance` or `issubclass`"
+
+    def leave_BooleanOperation(
+        self, original_node: cst.BooleanOperation, updated_node: cst.BooleanOperation
+    ) -> cst.CSTNode:
+        if self.matches_isinstance_issubclass_or_pattern(original_node):
+            if not self.filter_by_path_includes_or_excludes(
+                self.node_position(original_node)
+            ):
+                return updated_node
+
+            self.report_change(original_node)
+
+            elements = []
+            seen_values = set()
+            for call in extract_boolean_operands(updated_node, ensure_type=cst.Call):
+                class_tuple_arg_value = call.args[1].value
+                if isinstance(class_tuple_arg_value, cst.Tuple):
+                    arg_elements = class_tuple_arg_value.elements
+                else:
+                    arg_elements = (cst.Element(value=class_tuple_arg_value),)
+
+                for element in arg_elements:
+                    if (value := getattr(element.value, "value", None)) in seen_values:
+                        # If an element has a non-None evaluated value that has already been seen, continue to avoid duplicates
+                        continue
+                    if value is not None:
+                        seen_values.add(value)
+                    elements.append(element)
+
+            instance_arg = updated_node.left.args[0]
+            new_class_tuple_arg = cst.Arg(value=cst.Tuple(elements=elements))
+            return cst.Call(func=call.func, args=[instance_arg, new_class_tuple_arg])
+
+        return updated_node
+
+    def matches_isinstance_issubclass_or_pattern(
+        self, node: cst.BooleanOperation
+    ) -> bool:
+        # Match the pattern: isinstance(x, cls1) or isinstance(x, cls2) or isinstance(x, cls3) or ...
+        # and the same but with issubclass
+        args = [m.Arg(value=m.Name()), m.Arg(value=m.Name() | m.Tuple())]
```

Comment:
it's possible you can more cleanly do this entire match without libcst matchers and instead with a match / case statement. Check out `infer_expression_type` for a good example of how to write that.

Reply:
I'd love this comment to be addressed or at least push back explaining if it's not feasible.

Reply:
@clavedeluna Just noticed that I missed this comment. I see what you are saying about the case statement in `utils.infer_expression_type`. 

Should I rewrite it with a case statement now?

Reply:
Funny I just saw your made a comment about this at the exact same time I did. I  have an interview at 10, but I'll address this right after.

Reply:
yes, case statements are a bit nicer than isinstance statements especially with so many. let's see what it looks like

Reply:
I tried re-writing with a `match node: case ...` statement like in `infer_expression_type` and I think that that using `libcst.matchers.matches` here produces much more readable code since I am matching on node `shape` not attr values as described in the `libcst.matchers.matches` docs. 

`infer_expression_type` is matching a node by both shape and attr values which I believe is why a case statement make more sense there. 

To do this in `matches_isinstance_issubclass_or_pattern` would require a case for every cst.Node that matches the `isinstance_or` `issubclass_or` or `chained_or` shape and returning 
``` 
node.left.func.value == node.right.func.value  # Same function 
and node.left.args[0].value.value
== node.right.args[0].value.value  # Same first argument (instance)
``` 
for the `isinstance_or | issubclass_or` case 

and 
```
all(
                (
                    call.func.value == node.right.func.value  # Same function
                    and call.args[0].value.value
                    == node.right.args[0].value.value  # Same first argument (instance)
                )
                for call in extract_boolean_operands(node, ensure_type=cst.Call)
            )
```
for the `chained_or` case. 

As far as I can tell this wouldn't be possible to do like in `infer_expression_type` since the attr values are not known and a cst.Node to match with cannot be defined.

What do you think?

Reply:
Alright yeah I see what you mean, thanks for indulging me, we can definitely go with the existing implementation